pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,280 @@
+		ldapConfigs := &XmlLdapConfig{}
+		ldapSetting := unpackLdapSetting(d)
+
+		_, err := m.(*resty.Client).R().SetResult(&ldapConfigs).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+
+		matchedLdapSetting := LdapSetting{}
+		for _, iterLdapSetting := range ldapConfigs.Security.LdapSettings.LdapSettingArr {
+			if iterLdapSetting.Key == ldapSetting.Key {
+				matchedLdapSetting = iterLdapSetting
+				break
+			}
+		}
+		var ldapSettingClass = ignoreHclPredicate("class", "rclass", "manager_password")
+		packer := universalPack(
+			allHclPredicate(
+				ldapSettingClass, schemaHasKey(ldapSettingsSchema),
+			),
+		)
+		return diag.FromErr(packer(&matchedLdapSetting, d))
+	}
+
+	var resourceLdapSettingsUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedLdapSetting := unpackLdapSetting(d)
+
+		/* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		GET call structure has "security -> ldapSettings -> ldapSetting -> Array of ldapSetting config blocks".
+		PATCH call structure has "security -> ldapSettings -> Name/Key of ldap setting that is being patch -> config block of the ldapSetting being patched".
+		Since the Name/Key is dynamic string, following nested map of string structs are constructed to match the usage of PATCH call.
+		*/
+		var constructBody = map[string]map[string]map[string]LdapSetting{}
+		constructBody["security"] = map[string]map[string]LdapSetting{}
+		constructBody["security"]["ldapSettings"] = map[string]LdapSetting{}
+		constructBody["security"]["ldapSettings"][unpackedLdapSetting.Key] = unpackedLdapSetting
+		content, err := yaml.Marshal(&constructBody)
+
+		if err != nil {
+			return diag.Errorf("failed to marshal ldap settings during Update")
+		}
+
+		err = sendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		// we should only have one ldap setting resource, using same id
+		d.SetId(unpackedLdapSetting.Key)
+		return resourceLdapSettingsRead(ctx, d, m)
+	}
+
+	var resourceLdapSettingsDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		ldapConfigs := &XmlLdapConfig{}
+
+		rsrcLdapSetting := unpackLdapSetting(d)
+
+		response, err := m.(*resty.Client).R().Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		if response.IsError() {
+			return diag.Errorf("Got error response for API: /artifactory/api/system/configuration request during Read")
+		}
+
+		err = xml.Unmarshal(response.Body(), &ldapConfigs)
```

Comment:
Let's use Resty's auto unmarshaling here. If it works, then you also remove a set of error checking code block.